compiler/rustc_errors/src/lib.rs

```diff
@@ -1527,25 +1553,35 @@ impl DelayedDiagnostic {
         DelayedDiagnostic { inner: diagnostic, note: backtrace }
     }
 
-    fn decorate(mut self) -> Diagnostic {
+    fn decorate(mut self, dcx: &DiagCtxtInner) -> Diagnostic {
+        // FIXME: Cannot use `Diagnostic::subdiagnostic` which takes `DiagCtxt`, but it
+        // just uses `DiagCtxtInner` functions.
+        let subdiag_with = |diag: &mut Diagnostic, msg| {
```

Comment:
Can you pull this closure out further and use it for the FIXME cases above?

Reply:
I'm not sure I understand what that would look like.

Reply:
If you pulled this closure out into a method on `DiagCtxtInner`, then you could use it in `flush_delayed`, where the `FIXME` comment is.

Reply:
It's a little tricky to do this. I can't make the closure a method because the signature wouldn't match (`add_to_diagnostic_with` doesn't expect a `&Self`), it can't be a function which doesn't take `self` because it needs the reference to `DiagCtxtInner` (which we capture currently), and returning a closure that does match from this function is tricky because the closure captures the lifetime. I'm hoping to be able to remove the closure here in a follow-up anyway.

Reply:
ok